app/Http/Controllers/BookingAPIController.php

```diff
@@ -61,22 +62,59 @@ public function create(Request $request)
         }
 
         if (BookingService::dateConflict($room, $request->start, $request->end )) {
-                return response()->json([
-                    'message' => "The {$room->name} room is unavailable for the chosen dates."
-                ], 400);
+            return response()->json([
+                'message' => "The {$room->name} room is unavailable for the chosen dates."
+            ], 400);
         }
 
         $booking = new Booking();
-
         $booking->room_id = $request->room_id;
         $booking->customer = $request->customer;
         $booking->start = $request->start;
         $booking->end = $request->end;
-
         $booking->save();
 
         return response()->json([
             'message' => "Booking successfully created"
         ], 201);
     }
+
+         public function report()
+     {
+             $reportData = [];
+             // We add this at the top of the method in order to create an empty array (which
+             // we will fill with the required variables
+
+         $roomsWithBookings = Room::with('booking')
+         ->get()?->makeHidden(['rate', 'image', 'min_capacity', 'max_capacity', 'description', 'type_id', 'created_at',
+                 'updated_at', 'booking']);
```

Comment:
You can actually probably get away without using makeHidden here - since you only use this data to calculate the stats, you never actually return it.